File: palindrome/palindrome.js
'use strict';

function isPalindrome(inputString) {
  const cleanArray = inputString.toLowerCase().match(/[a-z\d]/g);
  const cleanArrayBackwards = [];
  if (!cleanArray || cleanArray.length < 2) return false;
  // Loop through cleanArray backwards and push each element onto our backwards array.
  for (let i = cleanArray.length - 1; i >= 0; i--) {
    cleanArrayBackwards.push(cleanArray[i]);
  }
  // Compare the original and backwards arrays. They should be the same if it's a palindrome.
  for (let j = 0; j < cleanArray.length; j++) {
    if (cleanArray[j] !== cleanArrayBackwards[j]) {
      return false;
    }
  }
  return true;
}

module.exports = { isPalindrome };

File: palindrome/palindrome.py
import sys, re


def isPalindrome(inputString):
    match = re.findall(r"[a-z0-9]+", inputString.lower())
    cleanString = "".join(match)
    if len(cleanString) < 2:
        return False
    elif cleanString == cleanString[::-1]:
        return True
    else:
        return False


def main():
    if len(sys.argv) != 2:
        print('usage: ./palindrome.py "String"')
        sys.exit(1)
    else:
        print(f"{sys.argv[1]} is Palindrome? - {isPalindrome(sys.argv[1])}")


if __name__ == "__main__":
    main()
